app/models/form_handler.rb

```diff
@@ -0,0 +1,27 @@
+class FormHandler
+  include Singleton
+
+  def initialize
+    @forms = {}
+    get_all_forms
```

Comment:
I think I'd be inclined to make `get_all_forms` more functional by just returning a {} rather than modifying @forms directly, and then possibly make it a `private` method to mark that it shouldn't really be called directly?